fabric-convention-tags-v1/src/datagen/java/net/fabricmc/fabric/impl/tag/convention/datagen/generators/ItemTagGenerator.java

```diff
@@ -0,0 +1,352 @@
+				.add(Items.LIGHT_GRAY_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.LIME_DYES)
+				.add(Items.LIME_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.MAGENTA_DYES)
+				.add(Items.MAGENTA_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.ORANGE_DYES)
+				.add(Items.ORANGE_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.PINK_DYES)
+				.add(Items.PINK_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.CYAN_DYES)
+				.add(Items.CYAN_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.GRAY_DYES)
+				.add(Items.GRAY_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.PURPLE_DYES)
+				.add(Items.PURPLE_DYE);
+	}
+
+	private void generateShulkerTag() {
+		getOrCreateTagBuilder(ConventionalItemTags.SHULKER_BOXES)
+				.add(Items.SHULKER_BOX)
+				.add(Items.BLUE_SHULKER_BOX)
+				.add(Items.BROWN_SHULKER_BOX)
+				.add(Items.CYAN_SHULKER_BOX)
+				.add(Items.GRAY_SHULKER_BOX)
+				.add(Items.GREEN_SHULKER_BOX)
+				.add(Items.LIGHT_BLUE_SHULKER_BOX)
+				.add(Items.LIGHT_GRAY_SHULKER_BOX)
+				.add(Items.LIME_SHULKER_BOX)
+				.add(Items.MAGENTA_SHULKER_BOX)
+				.add(Items.ORANGE_SHULKER_BOX)
+				.add(Items.PINK_SHULKER_BOX)
+				.add(Items.PURPLE_SHULKER_BOX)
+				.add(Items.RED_SHULKER_BOX)
+				.add(Items.WHITE_SHULKER_BOX)
+				.add(Items.YELLOW_SHULKER_BOX)
+				.add(Items.BLACK_SHULKER_BOX);
+	}
+
+	private void generateGlassTags() {
+		getOrCreateTagBuilder(ConventionalItemTags.GLASS_BLOCKS)
+				.add(Items.GLASS)
+				.add(Items.GRAY_STAINED_GLASS)
+				.add(Items.BLACK_STAINED_GLASS)
+				.add(Items.ORANGE_STAINED_GLASS)
+				.add(Items.BLUE_STAINED_GLASS)
+				.add(Items.BROWN_STAINED_GLASS)
+				.add(Items.CYAN_STAINED_GLASS)
+				.add(Items.GREEN_STAINED_GLASS)
+				.add(Items.LIGHT_BLUE_STAINED_GLASS)
+				.add(Items.LIGHT_GRAY_STAINED_GLASS)
+				.add(Items.LIME_STAINED_GLASS)
+				.add(Items.MAGENTA_STAINED_GLASS)
+				.add(Items.PINK_STAINED_GLASS)
+				.add(Items.PURPLE_STAINED_GLASS)
+				.add(Items.RED_STAINED_GLASS)
+				.add(Items.TINTED_GLASS)
+				.add(Items.WHITE_STAINED_GLASS)
+				.add(Items.YELLOW_STAINED_GLASS);
+		getOrCreateTagBuilder(ConventionalItemTags.GLASS_PANES)
+				.add(Items.GLASS_PANE)
+				.add(Items.GRAY_STAINED_GLASS_PANE)
+				.add(Items.BLACK_STAINED_GLASS_PANE)
+				.add(Items.ORANGE_STAINED_GLASS_PANE)
+				.add(Items.BLUE_STAINED_GLASS_PANE)
+				.add(Items.BROWN_STAINED_GLASS_PANE)
+				.add(Items.CYAN_STAINED_GLASS_PANE)
+				.add(Items.GREEN_STAINED_GLASS_PANE)
+				.add(Items.LIGHT_BLUE_STAINED_GLASS_PANE)
+				.add(Items.LIGHT_GRAY_STAINED_GLASS_PANE)
+				.add(Items.LIME_STAINED_GLASS_PANE)
+				.add(Items.MAGENTA_STAINED_GLASS_PANE)
+				.add(Items.PINK_STAINED_GLASS_PANE)
+				.add(Items.PURPLE_STAINED_GLASS_PANE)
+				.add(Items.RED_STAINED_GLASS_PANE)
+				.add(Items.WHITE_STAINED_GLASS_PANE)
+				.add(Items.YELLOW_STAINED_GLASS_PANE);
+	}
+
+	private void generateConsumableTags() {
+		getOrCreateTagBuilder(ConventionalItemTags.FOODS)
```

Comment:
In my opinion, this should most likely be replaced with an iteration over the item registry and include anything that has a food component to reduce the maintenance cost.

Reply:
I agree, there are a few places where this should be quite easy to do.

Reply:
Added.